### PR TITLE
Remove build listener usage

### PR DESCRIPTION
**Pull request by blindpirate**
So that we don't get deprecation warnings of configuration cache.

### CODE REVIEW

**Comment by blindpirate** [comment]
@bot-gradle test this

**Comment by blindpirate** [comment]
@bot-gradle test this

File: build-logic/lifecycle/src/main/kotlin/PrintStackTracesOnTimeoutBuildService.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    interface Params : BuildServiceParameters {
+        val timeoutMillis: Property<Long>
+        val projectDirectory: DirectoryProperty
+    }
+
+    private
+    lateinit var timer: Timer
+
+    fun startTimer() {
```
Comment (wolfs) [inline build-logic/lifecycle/src/main/kotlin/PrintStackTracesOnTimeoutBuildService.kt:39]: Can we start the timer in the constructor as well?

File: build-logic/lifecycle/src/main/kotlin/gradlebuild.lifecycle.gradle.kts [hunk 2]
```diff
@@ -51,24 +48,11 @@ tasks.registerEarlyFeedbackRootLifecycleTasks()
...
-            )
-        }
-        gradle.buildFinished {
-            timer.cancel()
-        }
+    if (BuildEnvironment.isCiServer) {
+        project.gradle.sharedServices.registerIfAbsent("printStackTracesOnTimeoutBuildService", PrintStackTracesOnTimeoutBuildService::class.java) {
+            parameters.timeoutMillis.set(determineTimeoutMillis())
+            parameters.projectDirectory.set(layout.projectDirectory)
+        }.get().startTimer()
```
Comment (wolfs) [inline build-logic/lifecycle/src/main/kotlin/gradlebuild.lifecycle.gradle.kts:55]: Do you know if we start the timer as well when we detect the accessors? This happens before we compile the build script which applies the plugin and that is why we had `project.name != "gradle-kotlin-dsl-accessors"` before.

Comment (blindpirate) [reply build-logic/lifecycle/src/main/kotlin/gradlebuild.lifecycle.gradle.kts]: I wasn't aware of that. I added it back in case.

**Comment by blindpirate** [comment]
@bot-gradle test this

File: build-logic/lifecycle/src/main/kotlin/PrintStackTracesOnTimeoutBuildService.kt (hunk 1 above)
Comment (blindpirate) [reply build-logic/lifecycle/src/main/kotlin/PrintStackTracesOnTimeoutBuildService.kt]: Done.

**Comment by blindpirate** [comment]
@bot-gradle test this

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Pull request merged**